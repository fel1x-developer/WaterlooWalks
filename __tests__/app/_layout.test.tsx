import { render } from "@testing-library/react-native";
import { Slot } from "expo-router";
import RootLayout from "../../app/_layout";

// Mock expo-router's Slot component
jest.mock("expo-router", () => ({
  Slot: jest.fn(() => {
    const React = require("react");
    const { View } = require("react-native");
    return React.createElement(View, { testID: "slot" });
  }),
}));

describe("RootLayout", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("should render without crashing", () => {
    const { toJSON } = render(<RootLayout />);
    expect(toJSON()).toBeTruthy();
  });

  it("should render the Slot component", () => {
    render(<RootLayout />);
    expect(Slot).toHaveBeenCalled();
  });

  it("should match snapshot", () => {
    const { toJSON } = render(<RootLayout />);
    expect(toJSON()).toMatchSnapshot();
  });
});
