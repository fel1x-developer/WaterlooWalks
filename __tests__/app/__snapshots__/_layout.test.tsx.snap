// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RootLayout should match snapshot 1`] = `
<View
  testID="slot"
/>
`;
